src/v2/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -174,19 +187,39 @@ export const ArtworkFilterContextProvider: React.FC<
     initialFilterState
   )
 
+  const [stagedArtworkFilterState, stage] = useReducer(artworkFilterReducer, {})
+
   // TODO: Consolidate this into additional reducer
   const [filterAggregations, setAggregations] = useState(aggregations)
   const [artworkCounts, setCounts] = useState(counts)
+  const [shouldStageFilterChanges, setShouldStageFilterChanges] = useState(
+    false
+  )
 
   useDeepCompareEffect(() => {
     if (onChange) {
       onChange(omit(artworkFilterState, ["reset"]))
     }
   }, [artworkFilterState])
 
+  // If in staged mode, return the staged filters for UI display
+  const currentlySelectedFilters = () => {
+    return shouldStageFilterChanges
+      ? stagedArtworkFilterState
+      : artworkFilterState
+  }
```

Comment:
We need to selectively read from either the staged or the real filters in order for the filter UI widgets  to render correctly